src/o365/base/DateAndPeriodBasedReport.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  CommandOption, CommandValidate
+} from '../../Command';
+import PeriodBasedReport, { OutputFileOptions } from './PeriodBasedReport';
+import * as path from 'path';
+import * as fs from 'fs';
+
+interface CommandArgs {
+  options: DateAndPeriodBasedOptions;
+}
+
+interface DateAndPeriodBasedOptions extends OutputFileOptions {
+  period?: string;
+  date?: string;
+}
+
+export default abstract class DateAndPeriodBasedReport extends PeriodBasedReport {
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const periodParameter: string = args.options.period ? `${this.usageEndPoint}(period='${encodeURIComponent(args.options.period)}')` : '';
+    const dateParameter: string = args.options.date ? `${this.usageEndPoint}(date=${encodeURIComponent(args.options.date)})` : '';
+    const endPoint: string = `${this.resource}/v1.0/reports/${(args.options.period ? periodParameter : dateParameter)}`;
+    this.executeReport(endPoint, cmd, args.options.output, args.options.outputFile, cb);
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.period = typeof args.options.period !== 'undefined';
```

Comment:
For `period` we should track the actual value. Since it's an enum, we could meaningfully aggregate it.